src/Schema.php

```diff
@@ -426,54 +426,48 @@ protected function getTableSequenceName(string $tableName): bool|float|int|strin
     {
-        $c = new ColumnSchema($column['column_name']);
-        $c->allowNull($column['nullable'] === 'Y');
-        $c->comment($column['column_comment'] ?? '');
-        $c->primaryKey((int) ($column['is_pk'] ?? 0) > 0);
-
-        $this->extractColumnType(
-            $c,
-            $column['data_type'],
-            $column['data_precision'],
-            $column['data_scale'],
-            $column['data_length']
-        );
-
-        $this->extractColumnSize(
-            $c,
-            $column['data_type'],
-            $column['data_precision'],
-            $column['data_scale'],
-            $column['data_length']
-        );
-
-        $c->phpType($this->getColumnPhpType($c));
-
-        if (!$c->isPrimaryKey()) {
-            if ($column['data_default'] === null) {
-                $c->defaultValue(null);
-            } else {
-                $defaultValue = $column['data_default'];
-
-                if ($defaultValue === 'CURRENT_TIMESTAMP' && $c->getType() === 'timestamp') {
-                    $c->defaultValue(new Expression('CURRENT_TIMESTAMP'));
-                } else {
-                    if (
-                        strlen($defaultValue) > 2
-                        && str_starts_with($defaultValue, "'")
-                        && str_ends_with($defaultValue, "'")
-                    ) {
-                        $defaultValue = substr($defaultValue, 1, -1);
-                    } else {
-                        $defaultValue = trim($defaultValue);
-                    }
-
-                    $c->defaultValue($c->phpTypecast($defaultValue));
-                }
-            }
-        }
+        $columnSchema = new ColumnSchema($column['column_name']);
+        $columnSchema->allowNull($column['nullable'] === 'Y');
+        $columnSchema->comment($column['column_comment']);
+        $columnSchema->primaryKey((bool) $column['is_pk']);
+        $columnSchema->size((int) $column['data_length']);
+        $columnSchema->precision($column['data_precision'] !== null ? (int) $column['data_precision'] : null);
+        $columnSchema->scale($column['data_scale'] !== null ? (int) $column['data_scale'] : null);
+        $columnSchema->dbType($column['data_type']);
+        $columnSchema->type($this->extractColumnType($columnSchema));
+        $columnSchema->phpType($this->getColumnPhpType($columnSchema));
+        $columnSchema->defaultValue($this->normalizeDefaultValue($column['data_default'], $columnSchema));
+
+        return $columnSchema;
+    }
 
-        return $c;
+    /**
+     * Converts column's default value according to {@see ColumnSchema::phpType} after retrieval from the database.
+     *
+     * @param string|null $defaultValue The default value retrieved from the database.
+     * @param ColumnSchemaInterface $columnSchema The column schema object.
+     *
+     * @return mixed The normalized default value.
+     *
+     * @psalm-suppress PossiblyNullArgument
```

Comment:
Why isn't using `@psalm-suppress PossiblyNullArgument` better to check that the argument is not null?

Reply:
To solve issue with psalm 4 tests with php 8.0
https://github.com/yiisoft/db-oracle/actions/runs/5470577402/jobs/9960814904#step:8:19
`Error: src/Schema.php:463:20: PossiblyNullArgument: Argument 1 of strlen cannot be null, possibly null value provided (see https://psalm.dev/078)`

Can be removed if do not run tests for psalm 4 with php 8.0

For psalm 5 with php 8.1 and 8.2 `@psalm-suppress PossiblyNullArgument` is not requered.